watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/dye_desalination/dye_desalination.py

```diff
@@ -9,102 +9,172 @@
+    Mixer,
+    Separator,
     Product,
     Translator,
+    MomentumMixingType,
 )
-import idaes.core.util.scaling as iscale
+from idaes.models.unit_models.separator import (
+    SplittingType,
+    EnergySplittingType,
+)
+
 from idaes.core import UnitModelCostingBlock
 
+from watertap.unit_models.pressure_exchanger import PressureExchanger
+from watertap.unit_models.pressure_changer import Pump
 from watertap.core.util.initialization import assert_degrees_of_freedom, check_solve
 
+import watertap.property_models.seawater_prop_pack as prop_SW
+from watertap.unit_models.reverse_osmosis_0D import (
+    ReverseOsmosis0D,
+    ConcentrationPolarizationType,
+    MassTransferCoefficient,
+    PressureChangeType,
+)
+from watertap.unit_models.reverse_osmosis_1D import ReverseOsmosis1D
+from watertap.costing.unit_models.dewatering import (
+    cost_centrifuge,
+    cost_filter_belt_press,
+    cost_filter_plate_press,
+)
+from watertap.property_models.multicomp_aq_sol_prop_pack import (
+    MCASParameterBlock,
+    DiffusivityCalculation,
+)
+from watertap.costing import MultiUnitModelCostingBlock
 from watertap.core.wt_database import Database
 import watertap.core.zero_order_properties as prop_ZO
 from watertap.unit_models.zero_order import (
     FeedZO,
     PumpElectricityZO,
     NanofiltrationZO,
+    SecondaryTreatmentWWTPZO,
 )
-from watertap.property_models.multicomp_aq_sol_prop_pack import (
-    MCASParameterBlock,
-    DiffusivityCalculation,
-)
-from watertap.unit_models.gac import (
-    GAC,
-)
-from watertap.costing.zero_order_costing import ZeroOrderCosting
+from watertap.unit_models.gac import GAC
 
-from watertap.core.util.model_diagnostics.infeasible import *
+from watertap.costing.zero_order_costing import ZeroOrderCosting
 from watertap.costing import WaterTAPCosting
 
+from idaes.core.util import DiagnosticsToolbox
 
 # Set up logger
 _log = idaeslog.getLogger(__name__)
 
 
 def main():
-    m = build(include_gac=False)
-
+    m = build(
+        RO_1D=True,
+        include_RO=True,
+        include_pretreatment=False,
+        include_dewatering=False,
+        include_gac=False,
+    )
     set_operating_conditions(m)
 
-    assert_degrees_of_freedom(m, 0)
     assert_units_consistent(m)
 
+    dt = DiagnosticsToolbox(m)
+    print("---Structural Issues---")
+    dt.report_structural_issues()
+
     initialize_system(m)
+    assert_degrees_of_freedom(m, 0)
 
     results = solve(m, checkpoint="solve flowsheet after initializing system")
 
-    display_results(m)
-
     add_costing(m)
-    assert_degrees_of_freedom(m, 0)
+    initialize_costing(m)
+    assert_degrees_of_freedom(m, 0)  # ensures problem is square
+
+    if hasattr(m.fs, "desalination"):
+        optimize_operation(m)  # unfixes specific variables for cost optimization
+    else:
+        pass
 
-    results = solve(m, checkpoint="solve flowsheet after costing")
+    solve(m, checkpoint="solve flowsheet after costing")
```

Comment:
So, I recall that `solve` uses `check_solve`, which would throw an InitializationError if `check_optimal_termination` returns False. However, an InitializationError wouldn't be appropriate here, so maybe we should just use assert_optimal_termination here. 

The original point of using the check_solve function was to help with troubleshooting.